core/src/HIP/Kokkos_HIP_ParallelReduce_Range.hpp

```diff
@@ -241,7 +241,7 @@ class ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits...>,
       dim3 block(1, block_size, 1);
       // use a slightly less constrained, but still well bounded limit for
       // scratch
-      int nblocks = (nwork + block.y - 1) / block.y;
+      index_type nblocks = (nwork + block.y - 1) / block.y;
```

Comment:
What's the deal with this change and the conversion below?

Reply:
Now that we favor smaller bloc size, there are cases where we overflow `nblock`. The conversion is because the two arguments of `std::min` need to be the same type.